core/matrix/batch_struct.hpp

```diff
@@ -22,14 +22,14 @@ namespace csr {
 /**
  * Encapsulates one matrix from a batch of csr matrices.
  */
-template <typename ValueType, typename IndexType>
+template <typename ValueType, typename IndexType = const int32>
```

Comment:
does it need to be `const int32`?
My personal preference is `int32` and use `const IndexType` when it is necessary

Reply:
It was necessary to add a default parameter, because the CSR and ELL have two template arguments, while Dense only has one. So, to handle them the same in the template instantiation, I had to use either add another template argument to Dense (which will not be used), or add a default argument to CSR and ELL. Since CSR and ELL already expect to only use `int32` as index type, I chose that.
Additionally, it had to be `const`, because the default argument has to match how the solver functions will be called. Otherwise, there will be a mismatch between the instantiated functions and what is actually called, leading to linker errors.

Reply:
Any more comments? If not, I will merge this today, or tomorrow morning.